aten/src/ATen/native/Blas.cpp

```diff
@@ -91,4 +94,31 @@ Tensor mv(const Tensor &self, const Tensor &vec) {
   return native::mv_out(result, self, vec);
 }
 
+Tensor dot(const Tensor &self, const Tensor &other){
+  at::NoNamesGuard guard;
+
```

Comment:
should you also check that tensors have same dtype?

Reply:
It does error out. 
But I guess explicit check with a nicer error message would be better.
```python
>>> import torch
>>> a = torch.randn(3)
>>> b = torch.randn(3, dtype=torch.double)
>>> a.dot(b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: expected scalar type Float but found Double
```

Reply:
Hm, do you know where is the check happening?

Reply:
Not sure but I think it happens in dispatch, while trying to get ptr to self and other, dispatch expects them to be similar type.

Will add an explicit check.

Reply:
Ah, you are right, it's most likely `other.data_ptr<scalar_t>()`.

Reply:
Have added an explicit check.